enhancements/sig-architecture/229-manifestworkreplicaset-rollback/README.md

```diff
@@ -0,0 +1,425 @@
+    message: "New revision mwrs-XXXXXXX is successfully progressed"
+    reason: NewManifestWorkAvailable
+    status: "True"
+    type: Progressing
+
+  placementSummary:
+  - availableDecisionGroups: 1 (2 / 2 clusters applied)
+    name: placement1
+    summary:
+      # (NEW) updated: count of clusters with the update revision applied
+      updated: 2
+      Applied: 2
+      available: 2
+      degraded: 0
+      progressing: 0
+      total: 2
+  summary:
+    # (NEW) updated: count of clusters with the update revision applied
+    updated: 2
+    Applied: 2
+    available: 2
+    total: 2
+```
+
+#### API Status
+
+This enhancement introduces `RolloutDegraded` reason to `PlacementRollout` condition type to make the current rollout status clear. This enhancement also added `Progressing` condition type for updating the current rollout progress.
+
+| Condition Type | Reasons | Description |
+| --- | --- | --- |
+| PlacementVerified | AsExpected, PlacementDecisionNotFound, PlacementDecisionEmpty, NotAsExpected | Indicates if Placement is valid |
+| ManifestworkApplied | AsExpected, NotAsExpected, Processing | A ManifestWork has been created in each cluster defined by PlacementDecision |
+| PlacementRollOut | Progressing, Complete, `(NEW)` RolloutDegraded | Indicates if RollOut Strategy is complete |
+| `(NEW)` Progressing | NewRevisionCreated, FoundNewRevision, NewManifestWorkAvailable, RolloutAborted, ProgressDeadlineExceeded | The rollout is progressing. Progress for a rollout is considered when a new manifest is created or adopted, when new manifestwork rolls out |
+
+
+### ManifestWork API Object
+
+To track the revision of manifestwork, this enhancement will introduce the `work.open-cluster-management.io/controller-hash` label. The hash can be used to identify the ControllerRevision history and filter `ManifestWork` resources.
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata:
+  name: <MWRS_Name>
+  namespace: <cluster name>
+  labels:
+    # (NEW) controller-hash is to distinguish between old and new
+    # Manifestworks during Manifestwork template updates.
+    work.open-cluster-management.io/controller-hash: <hash>
+
+    # existing labels.
+    work.open-cluster-management.io/manifestworkreplicaset: default.mwrset-1
+    work.open-cluster-management.io/placementname: placement1
+spec: 
+  ...
+```
+
+### `Abort` vs `Rollback`
+
+`Abort` is used to **cancel** the current rollout. When aborting the current rollout, spec is not changed, but `status.abort` is set to `true`, which will set `status.abortedTime`. then loading the older revision and then apply `.spec.manifestWorkTemplate` of the old revision to all clusters at once.
+
+On the other hand, `rollback` is the explicit action. If user wants to roll back to the older revision, the user (or cli) can find the revision from the list of ControllerRevision resources and apply the older manifestTemplate to update the `.spec.manifestWorkTemplate`.
```

Comment:
Does this mean the user can not use Revision to manual rollout? but need to copy and update the `.spec.manifestWorkTemplate`.